charger/ocpp.go

```diff
@@ -375,21 +389,34 @@ func (c *OCPP) updatePeriod(current float64, phases int) error {
 
 	current = math.Trunc(10*current) / 10
 
-	err = c.setChargingProfile(c.connector, getTxChargingProfile(current, phases, txn))
+	err = c.setChargingProfile(c.connector, c.getTxChargingProfile(current, txn))
 	if err != nil {
 		err = fmt.Errorf("set charging profile: %w", err)
 	}
 
 	return err
 }
 
-func getTxChargingProfile(current float64, phases, transactionId int) *types.ChargingProfile {
-	period := types.NewChargingSchedulePeriod(0, current)
+func (c *OCPP) getTxChargingProfile(current float64, transactionId int) *types.ChargingProfile {
+	var phases = c.phases
+	var value float64
+	if c.chargingRateUnit == types.ChargingRateUnitWatts {
+		voltage := 230.0
+		// Get number of currently active phases from Loadpoint
```

Comment:
Check if lp is really non-nil or exit

Reply:
I could log a warning and fall back to c.phases?

Reply:
Where would you get c.phases from? It would be 0. Warning is not necessary as this can only be temporary.

Reply:
Ok. Hier fehlt mir Go-Wissen. Wie komme ich im Fehlerfall aus der Funktion heraus, ohne weiter oben NullPointerExceptions (weiß nicht, wie das in Go heißt) zu provozieren? Hints welcome.